index.bs

```diff
@@ -522,6 +522,158 @@ See also:
 * [Security and privacy are essential](https://www.w3.org/2001/tag/doc/ethical-web-principles/#privacy)
 * [What data does this specification expose to an origin?](https://www.w3.org/TR/security-privacy-questionnaire/#underlying-platform-data)
 
+<h3 id="support-non-fully-active">Support non-"fully active" documents</h3>
+
+After a user navigated away from a document,
+the document might stay around in a non-"[=Document/fully active=]" state,
+and might be reused when the user navigates back to the entry holding the document.
+In the past, many APIs have missed specifying support for non-"[=Document/fully active=]" documents,
+making them hard to support in various user agents' implementation of "back/forward cache" (BFCache).
+
+To avoid this happening with your API,
+you should specify support for non-"[=Document/fully active=]" documents by following these guidelines:
+
+<h4 id="gate-fully-active">Gate actions with "[=Document/fully active=]" checks</h4>
+
+When performing actions that might update the state of a document,
+be aware that the document might not be "[=Document/fully active=]"
+and is considered as "non-existent" from the user's perspective.
+This means they should not receive updates or perform actions.
+
+In many cases,
+anything that happens while the document is not "[=Document/fully active=]"
+should be treated as if it never happened,
+and not "queue" the updates and send them after the document becomes fully activated.
+If it makes sense to send the latest state update (and only the latest state,
+not a queue of state changes) to the document after it becomes "[=Document/fully active=]" again,
+consider the "listen for changes" pattern below.
+<div class="example">
+  APIs that periodically send information updates,
+  such as Geolocation API's "[watch position](https://w3c.github.io/geolocation-api/#watchposition-method)"
+  should not send updates if the document is no longer fully active.
+  It also should not queue those updates to arrive later,
+  and only resume sending updates when the document becomes active with the latest location.
+</div>
+
+<h4 id="listen-fully-active">Listen for changes to "[=Document/fully active=]" status</h4>
+
+When a document goes from "[=Document/fully active=]" to non-"[=Document/fully active=]",
+it should be treated similarly to the way discarded documents are treated.
+The document must not retain exclusive access to shared resources
+and must ensure that no new requests are issued
+and that connections that allow for new incoming requests are terminated. 
+When a document goes from non-"[=Document/fully active=]" to "[=Document/fully active=]" again,
+it can restore connections if appropriate.
+  
+While web authors can manually do cleanup (e.g. release the resources, sever connections)
+from within the "pagehide" event and restore them from the "pageshow" event themselves,
+doing this automatically from the API design would reduce a lot of overhead.
```

Comment:
I don't think "reduce overhead" is very helpful here. Maybe "is better for web developers" or "is more likely to lead to well-functioning web applications" or something like that?

Reply:
OK, added this and also mentioned "bfcache by default" a little bit